package.json

```diff
@@ -164,7 +164,7 @@
   },
   "packageManager": "pnpm@9.6.0",
   "engines": {
-    "pnpm": "9.6.0"
+    "pnpm": "^9.0.0"
```

Comment:
Per https://github.com/mui/mui-toolpad/pull/3396#discussion_r1564881410, it looks like we should revert, no?

Reply:
The Twitter thread in the first post has the reasoning. Restricting the major version makes sense, but I don't see why patch/minors matter when the same major works correctly, it makes it annoying to work on the project by constantly needing to ensure the pnpm version is synchronized.  Unless there's a recommended tool to make that automatic

Reply:
> Unless there's a recommended tool to make that automatic

I believe corepack does this automatically.

Reply:
I did try corepack a ~month ago after the change, but it was causing issues again recently for some reason (it's still listed as experimental). I figured this wasn't worth the trouble to deal with because it wastes time trying to work around it; having the minors/patches of your pnpm version be different locally won't cause issues in practice.

It's similar to transitive deps being permissive with the caret versioning of a major, which in practice works fine until a specific bug comes up and an update is necessary, which is rare - especially with a package manager.

Reply:
Interesting, if corepack fails to function correctly, it's pain indeed 🙃. Not sure what we should do then. I think I did my part, I raised this opportunity, then yeah, maybe we should ignore it. 

cc @Janpot for awareness that not all community contributors might have corepack working.

Reply:
What's the reasoning? That Twitter post just says "pro-tip: if you see this in package.json, delete it".

Which issues are you seeing enabling corepack?

Reply:
The `packageManager` field is what CI uses, so that remains reproducible as it's kept pinned to the version.  It's only locally that the "engines" field becomes annoying. 

In replies he mentions:
> "dont make me install a different version or node and npm when my current env works for your project"

It's good that the brand new pnpm release 9.7.0 seems to automatically fix this issue though. So maybe this isn't necessary anymore anyway.

Reply:
> The packageManager field is what CI uses, so that remains reproducible as it's kept pinned to the version. It's only locally that the "engines" field becomes annoying.

Yep, I see it more as a second layer of defence, to prevent anyone from configuring CI with the wrong package manager.

> "dont make me install a different version or node and npm when my current env works for your project"

This may be somewhat ok for external contributors, but to me it's not really about whether "it works" or not. For maintainers that have access to project/company secrets I think we should exclude outdated runtimes and package managers. Just to reduce potential attack surface through known security issues. Same reason why we wouldn't really want our team members to use outdated browsers to access company services.

Reply:
Alright, then I think we can: keep things like this, at one point once corepack is stable and it works in James' local environment we revert this PR. If we have other contributors struggling with corepack, we can reconsider, so far, it seems a one-off instance.

Reply:
For the record, I haven't had any issues with Corepack so far

Reply:
Oh well: https://github.com/nodejs/package-maintenance/pull/606, no more corepack in node by default 🙃

Reply:
Agree, it's still ok, but it would have been so nice for it to "just work".

Reply:
It should be fine to loosen the `engines.pnpm` restriction to the first one that supports the `packageManager` field. i.e. [`>=9.7`](https://github.com/pnpm/pnpm/releases/tag/v9.7.0) + add `manage-package-manager-versions`.